### PR TITLE
modify sm jit workflow to only touch files and not download them

### PR DESCRIPTION
**Pull request by ayushkamat**
Signed-off-by: Ayush Kamat <ayush@latch.bio>

### CODE REVIEW

File: latch/types/directory.py [hunk 1]
```diff
@@ -93,6 +110,8 @@ def __init__(
         else:
             self.path = str(path)
 
+        self._path_witness = False
```
Comment (maximsmol) [inline latch/types/directory.py:113]: any difference between this and `self.path is not None`?

Comment (maximsmol) [reply latch/types/directory.py:113]:
```suggestion
        self._path_generated = False
```

File: latch/types/directory.py [hunk 2]
```diff
@@ -165,6 +196,36 @@ def iterdir(self) -> List[Union[LatchFile, "LatchDir"]]:
 
         return ret
 
+    def touch(self):  # fixme(ayush): better name
```
Comment (maximsmol) [inline latch/types/directory.py:199]:
```suggestion
    def _create_impostors(self):
```

File: latch/types/directory.py [hunk 3]
```diff
@@ -165,6 +196,36 @@ def iterdir(self) -> List[Union[LatchFile, "LatchDir"]]:
...
+            """),
+            {"path": self._remote_directory},
+        )["ldataResolvePathData"]
+
+        if res is None:
+            # todo(ayush): proper error message + exit
+            raise FlyteUserException(f"No directory at {self._remote_directory}")
+
+        for x in res["finalLinkTarget"]["descendants"]["nodes"]:
+            p = Path(self.path) / x["relPath"]
```
Comment (maximsmol) [inline latch/types/directory.py:224]: put `Path(self.path)` in a variable before the loop

File: latch/types/file.py [hunk 4]
```diff
@@ -73,6 +75,8 @@ def __init__(
         else:
             self.path = str(path)
 
+        self._path_witness = False
```
Comment (maximsmol) [inline latch/types/file.py:78]: same as in dir

File: latch/types/file.py [hunk 5]
```diff
@@ -116,6 +121,24 @@ def downloader():
...
+            return
+
+        ctx = FlyteContextManager.current_context()
+        if ctx is None:
+            return
+
+        self.path = ctx.file_access.get_random_local_path(hint)
+        self._path_witness = True
+
+    def touch(self):
```
Comment (maximsmol) [inline latch/types/file.py:135]: same as in dir

**Review by maximsmol — CHANGES_REQUESTED** [review]

File: latch/types/directory.py (hunk 1 above)
Comment (ayushkamat) [reply latch/types/directory.py:113]: `self.path` is set in `__init__` (this needs to change at some point)

**Pull request merged**